helpers/callBound.js

```diff
@@ -1,14 +1,11 @@
 'use strict';
 
 var GetIntrinsic = require('../GetIntrinsic');
-
 var callBind = require('./callBind');
 
-var $indexOf = callBind(GetIntrinsic('String.prototype.indexOf'));
-
 module.exports = function callBoundIntrinsic(name, allowMissing) {
 	var intrinsic = GetIntrinsic(name, !!allowMissing);
-	if (typeof intrinsic === 'function' && $indexOf(name, '.prototype.')) {
+	if (typeof intrinsic === 'function') {
```

Comment:
This definitely can't be right. I see the bug before - that it's always binding - but the fix definitely shouldn't be "always bind".

If this:
```suggestion
	if (typeof intrinsic === 'function' && $indexOf(name, '.prototype.') >= 0) {
```
breaks something else, let's fix that something else so it works properly with this change.

Reply:
This change:
```suggestion
	if (typeof intrinsic === 'function' && ($indexOf(name, '.prototype.') >= 0 || $indexOf(name, 'Promise.') === 0 || $indexOf(name, '%Promise.') === 0)) {
```
seems to work with all of my projects, and everything I can find via github code search. Are there dependents you're aware of that would break with this suggestion?

Reply:
ping @ExE-Boss; thoughts on this suggestion?

Reply:
Well, there’s other non‑prototype methods that need to support being `callBound`ed, such as the `%TypedArray%` statics: <https://github.com/nodejs/node/blob/b49145f795658c7187a94897d31f665c2fbff471/lib/internal/per_context/primordials.js#L230-L241>

Reply:
OK, I handle Promise already, I can add the TypedArrays to the list. Anything else?

Reply:
Most of the built‑in static methods that perform object creation support constructors being passed as the `this` binding, and the custom constructor will then be used instead of the original.

Reply:
The ones that support a custom receiver aren't what matters - only the ones that will fail when provided *no* receiver, like the Promise ones (and perhaps the TypedArray ones), would need call-binding.